### PR TITLE
Fix #10727: Avoid segfault on null strings in std.file.rename (stable…

### LINKED ISSUE #10727: std.file.rename segfaults on null string on Windows
```D
import std.file;
void main()
{
    string f = null;
    rename("", f);
    rename(f, "");
}
```

This segfaults on Windows. When trying to create an Exception message string, it calls `wcslen(null)`:

https://github.com/dlang/phobos/blob/94903c884a20cdb5b9e341ced6849f989cae5334/std/file.d#L986

The C string is `null` because of:

https://github.com/dlang/phobos/blob/94903c884a20cdb5b9e341ced6849f989cae5334/std/internal/cstring.d#L88-L96

### PR DESCRIPTION
**Pull request by Mahipalsinh00**
Fixes #10727  
Complements #10736 (adds unit tests)

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -983,10 +995,10 @@ private void renameImpl(scope const(char)[] f, scope const(char)[] t,
             import std.conv : to, text;
 
             if (!f)
-                f = fromz ? to!(typeof(f))(fromz[0 .. wcslen(fromz)]) : "(null)";
+                f = fromz is null ? null : to!(typeof(f))(fromz[0 .. wcslen(fromz)]);
```
Comment (CyberShadow) [inline std/file.d:998]: This does not seem like an improvement.

Comment (Mahipalsinh00) [reply std/file.d:998]: @CyberShadow Thank you for taking the time to review this!  I appreciate your insight, and I’ll rework the fix to better align with the intended behavior.
 -I’ve also added unit tests
Would you kindly share any specific guidance or edge cases I should prioritize? I’m happy to iterate further based on your feedback!

**Review by 0xEAB — CHANGES_REQUESTED** [review]
Thanks you for your work!
Unfortunately, the added unittest is unsuitable to verify the effectiveness of your changes. As demonstrated in #10770 it also passes with the unchanged code currently on ~stable.

Please add a unittest that actually outlines the benefit(s) of your change.

**Comment by 0xEAB** [comment]
> As demonstrated in #10770 it also passes with the unchanged code currently on ~stable.

In other words: This implies that either #10727 has already been fixed or your unittest is not suitable to test for a fix of the reported issue. (To be honest, I suspect the former may be the case.)

File: std/file.d [hunk 2]
```diff
@@ -969,6 +969,18 @@ if (isConvertibleToString!RF || isConvertibleToString!RT)
     assert(t2.readText == "2");
 }
 
+unittest
```
Comment (0xEAB) [inline std/file.d:972]:
```
std/file.d(972:1)[warn]: A unittest should be annotated with at least @safe or @system
```

File: std/file.d (hunk 1 above)
Comment (CyberShadow) [reply std/file.d:998]: @Mahipalsinh00 Could you explain why you think the new version of the code is better?

Comment (Mahipalsinh00) [reply std/file.d:998]: @CyberShadow In new version of code :
1) Explicit null check:
fromz is null is clear and unambiguous, avoiding implicit truthy/falsy behavior.

2) Preserves null:
If fromz is null, f also becomes null. This is more semantically accurate and safer for further processing (e.g., you can check if (f is null) later).

3) Avoids mixing types:
The first version returns a literal string "(null)", which could create type inconsistencies or confuse downstream logic if you're expecting f to be null-checkable.

Conclusion:  The new version is better because it is cleaner, more type-safe, and logically accurate.

**Comment by CyberShadow** [comment]
It looks like @Mahipalsinh00 is using an LLM - at least for writing comments (see https://github.com/dlang/phobos/pull/10769#discussion_r2071436090).

I'm going to close this PR because:

1. the author is not acting in good faith
2. it is superseded by https://github.com/dlang/phobos/pull/10770.

@Mahipalsinh00 Thank you for submitting the unit test. Please see my comment on a similar contribution [here](https://github.com/dlang/dlang.org/pull/4214#issuecomment-2772117877) regarding use of LLMs.

File: std/file.d (hunk 1 above)
Comment (CyberShadow) [reply std/file.d:998]: It's quite clear from the above comment that you didn't even read the surrounding code, and it looks like you just posted a comment generated by an LLM. This is not appreciated, please do not do this if you plan on continuing to contribute to this project.

Comment (Mahipalsinh00) [reply std/file.d:998]: I am  really sorry. I will take a note to not use any LLM  in the future.
I posted a comment generated by an LLM because I think it will explain in better way. 
And I will apologize for that
